locator/token_metadata.hh

```diff
@@ -70,6 +70,16 @@ struct host_id_or_endpoint {
 
 class token_metadata_impl;
 
+// While updating topology through Raft, a transition is required to shift
+// read requests from old to new endpoints. This enum establishes two related values:
+//     read_old - no read_endpoints will be stored in update_pending_ranges,
+//                all reads are directed to normal endpoints;
+//     read_new - initiates update_pending_ranges to calculate and store new ranges for read requests.
+enum class topology_change_stage {
```

Comment:
This will lead to the refactoring of `service::topology`, creating a coupling with `locator::topology` just to reuse this enum. This doesn't seem worth it to me.

Maybe we should unite these two topologies (I'm not sure about that), but it sounds like a separate task with its own objective and value.

Reply:
updated